### PR TITLE
HS-1564: Automation: UI: Flows

### PR DESCRIPTION
**Pull request by maximbrener**
This PR contains:

1. support enabling and disabling welcome wizard per test. By default any tests with @cloud should skip the wizard
2. left nav bar issue when its not hiding
3. tests are using env variables: 
            CLOUD_MINION_GATEWAY_HOST
            CLOUD_INSTANCE_URL
            CLOUD_USERNAME
            CLOUD_PASSWORD
            IS_CLOUD_TESTING_TARGET
4. basic flows test:
        - create location
        - download cert from location page
        - start minion with SNMP and UDPGen
        - create new discovery from discovery page
        - check node discovered and monitored
        - check flow data visible on home page
        - check flow data visible in the flow page

### CODE REVIEW

**Comment by agalue** [comment]
As a suggestion, and because you use `udpgen`, and the total number of flows is predictable, it would be great to validate if the Metrics Processor gets all the flows that have been sent (which doesn't happen based on my extensive tests, a problem I reported months ago).

**Comment by maximbrener** [comment]
> 

We absoluttly should have that covered. This PR is about basic sanity of flows through UI only. To check metric processor we need smehow to be able to access metics there (if test runs against cloud) or to spin up within pipeline all the missing flows components. Thoughts?

**Comment by agalue** [comment]
Certainly, Lokahi won't forward flows, as there are no Flow Services without the Cloud environment. However, the flow data should reach the Metrics Processor when Minion sends them regardless of that. There, there is a Prometheus Metric called `flow_sample_count_total`, for instance:
```bash
❯ curl "http://localhost:8080/actuator/prometheus" 2>/dev/null | grep "^flow_sample_count_total"
flow_sample_count_total{tenant="3c0c5318-bb83-4679-92b8-c0753cb72680",} 152.0
```
And if everything is working (which isn't currently), that should match the JMX metric on Minion called `parsers.Netflow-9-Parser.recordsCompleted.Count`. This metric must check the number of Flows sent by `udpgen`. As this tool sends one Flow Record per FlowSet when using Netflow 9, the number of packets should match the number of flows set (Netflow 5 only sends one flow at a time, which is equivalent).

**Comment by maximbrener** [comment]
> Certainly, Lokahi won't forward flows, as there are no Flow Services without the Cloud environment. However, the flow data should reach the Metrics Processor when Minion sends them regardless of that. There, there is a Prometheus Metric called `flow_sample_count_total`, for instance:
> 
> ```shell
> ❯ curl "http://localhost:8080/actuator/prometheus" 2>/dev/null | grep "^flow_sample_count_total"
> flow_sample_count_total{tenant="3c0c5318-bb83-4679-92b8-c0753cb72680",} 152.0
> ```
> 
> And if everything is working (which isn't currently), that should match the JMX metric on Minion called `parsers.Netflow-9-Parser.recordsCompleted.Count`. This metric must check the number of Flows sent by `udpgen`. As this tool sends one Flow Record per FlowSet when using Netflow 9, the number of packets should match the number of flows set (Netflow 5 only sends one flow at a time, which is equivalent).

Sounds good. We will keep increrasing our coverage as we go, this one seems as good next step for flows. Will talk to you later for more details. Thanks.

File: test-automation/ui-tests/src/test/resources/cloud-features/flows.feature [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Given Navigate to the "locations" through the left panel
+    Given location "Location Flow Testing" created
+    Then click on 'Download Certificate' button, get certificate and password for minion "minion_test_id_2" and start minion using "docker-compose-cloud.yaml"
+    Then check "Location Flow Testing" location exists
+    Then click on location "Location Flow Testing"
+    Then check minion "minion_test_id_2" exists
+    Then Navigate to the "discovery" through the left panel
+    Then add SNMP discovery "Flows Testing Discovery" for location "Location Flow Testing" with IP "172.31.10.10"
+    Then Navigate to the "inventory" through the left panel
+    Then check node with system name "automated-testing-sysName" discovered successfully
```
Comment (maximbrener) [inline test-automation/ui-tests/src/test/resources/cloud-features/flows.feature:20]: this step currently fails due to https://opennms.atlassian.net/browse/HS-1821.

File: test-automation/ui-tests/src/test/resources/cloud-features/flows.feature [hunk 2]
```diff
@@ -0,0 +1,27 @@
+@cloud
+Feature: NetFlows basic function
+
+  Scenario: User sees empty 'Flows' chart before data was sent
+    Given No netflow data was sent
+    And sees "No applications data was found in last 24 hours.." subtitle in the 'Top 10 Applications' chart
+    Then click on 'Flows' link
+    And sees 'No Data' in the flows table
+
+  Scenario: User sees 'Top 10 Applications' chart after flows data came to the instance
```
Comment (bouff) [inline test-automation/ui-tests/src/test/resources/cloud-features/flows.feature:10]: Nit: suggest replacing with `flows data sent to the instance`

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/cloud/DiscoveryPage.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+
+import static com.codeborne.selenide.Selectors.withText;
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.$$;
+
+public class DiscoveryPage {
+
+    private static final SelenideElement addDiscoveryButton =  $("[class='btn hover focus btn-primary has-icon']");
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/cloud/DiscoveryPage.java:39]: Nit: several files in this PR have dangling spaces.  Thinking we should implement a linter task we can run locally.  (And ultimately have this running in CI as well).

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/cloud/DiscoveryPage.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    private static final SelenideElement SNMPRadioButton =  $("[class='radio hover focus']");
+    private static final SelenideElement discoveryNameInputField =  $("[class='feather-input']");
+    private static final SelenideElement discoveryLocationNameInputField = $$("[class='feather-autocomplete-input']").get(1);
+    private static final SelenideElement discoveryLocationNamesDropdown = $("[data-ref-id='feather-autocomplete-menu-container-dropdown']");
+    private static final SelenideElement discoveryIPInputField = $("[id='contentEditable_1']");
+    private static final SelenideElement saveDiscoveryButton = $("[type='submit']");
+    private static final SelenideElement addAnotherDiscoveryButton = $("[class='btn hover focus btn-text has-icon']");
+    public static void selectICMP_SNMP() {
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/pages/cloud/DiscoveryPage.java:47]: Nit: missing newline here.

File: test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/CucumberHooks.java [hunk 5]
```diff
@@ -61,17 +62,15 @@ public class CucumberHooks {
     private static final String ADMIN_DEFAULT_USERNAME = "admin";
     private static final String ADMIN_DEFAULT_PASSWORD = "admin";
     private static final String CLOUD_MINION_GATEWAY_HOST_PORT = "443";
-    private static final String CLOUD_MINION_GATEWAY_HOST_DEFAULT = "minion.onms-fb-dev.dev.nonprod.dataservice.opennms.com";
+    private static final String CLOUD_MINION_GATEWAY_HOST = "CLOUD_MINION_GATEWAY_HOST";
```
Comment (bouff) [inline test-automation/ui-tests/src/test/java/org/opennms/horizon/systemtests/CucumberHooks.java:65]: Nit: recommend adding `ENV` in naming constants for the environment variable names.

**Review by bouff — APPROVED** [review]
LGTM!

**Pull request merged**